.gitbook/contents/hc/4_deploy-smart-contract.md

```diff
@@ -0,0 +1,62 @@
+# Deploy the Smart Contract
+
+With a few adjustments to the `deploy-local.py` script, we can easily deploy our newly created smart contract.
+
+First, we need to configure the script to connect to our L1 and L2 networks:
+
+```solidity
+l1 = Web3(Web3.HTTPProvider("http://127.0.0.1:8545"))
+assert(l1.is_connected)
+l1.middleware_onion.inject(geth_poa_middleware, layer = 0)
+
+l2 = Web3(Web3.HTTPProvider("http://127.0.0.1:9545"))
+assert(l2.is_connected)
+l2.middleware_onion.inject(geth_poa_middleware, layer = 0)
+```
+
+Next, we load the contract using the `loadContract()` function:
+
+```solidity
+TC = loadContract(w3, "TestCounter", path_prefix + "test/TestCounter.sol")
+```
+
+Here `path_prefix` indicates where the contract is located.
+
+After loading the contract, we can deploy it as follows:
+
+```solidity
+epAddr = deploy2("EntryPoint", EP.constructor(), 0)
```

Comment:
maybe lets add a note here that deploy2 is just a function that deploys the smart contract with the deploy2 opcode for deterministic smart contract addresses.